### PR TITLE
Improved 2D meshing

### PR DESCRIPTION
**Pull request by simbilod**
Key improvements to 2D GMSH plugin:
* Now calls a more broadly useful shapely <--> gmsh converter (meshtracker), using an OrderedDict for shape priority
* Cleanup after user: merging polygons, rounding coordinates, simplifying polygons
* Seamless handling of lateral interfaces
* Seamless handling of vertical interfaces
* `uz_xsection_mesh` can return a 2D simulation cross-section in the vertical plane from a component along an arbitrary (x1,y1), (x2,y2) line, e.g. :

image

* `xy_xsection_mesh` can return a 2D simulation cross-section in the xy-plane from a component for given `z`-value, e.g. :

image

* Surfaces identified by layername, ability to manually add lines, or to use automatically-tagged interfaces (very important to set boundary conditions):

image

* Coarse refinement within a surface or away from an edge/interface

TODO:

- [x] Finish adding tests
- [ ] Finish 3D version
- [ ] Add mesh refinement
- [ ] Fix bugs as they come up

### CODE REVIEW

**Comment by joamatab** [comment]
Thank you Simon, it looks great

there are still some flake8 issues
https://github.com/gdsfactory/gdsfactory/actions/runs/3415421287/jobs/5684532638

**Comment by simbilod** [comment]
Hmm pre-commit was passing on my end, I'll take another look!

**Comment by simbilod** [comment]
@joamatab what about now?

**Comment by simbilod** [comment]
uz can now do sidewalls

image

File: gdsfactory/tech.py [hunk 1]
```diff
@@ -189,38 +193,19 @@ def get_klayout_3d_script(self) -> str:
 
 def get_layer_stack_generic(
     thickness_wg: float = 220 * nm,
-    thickness_slab_deep_etch: float = 90 * nm,
     thickness_clad: float = 3.0,
     thickness_nitride: float = 350 * nm,
-    thickness_ge: float = 500 * nm,
     gap_silicon_to_nitride: float = 100 * nm,
-    zmin_heater: float = 1.1,
```
Comment (joamatab) [inline gdsfactory/tech.py:197]: Why remove this?

File: gdsfactory/tech.py [hunk 2]
```diff
@@ -415,6 +398,19 @@ def assert_callable(function):
         )
 
 
+class Tech(BaseModel):
```
Comment (joamatab) [inline gdsfactory/tech.py:401]: This was deleted, why bring it back?

File: gdsfactory/tech.py (hunk 1 above)
Comment (simbilod) [reply gdsfactory/tech.py]: tech.py has been updated, good catches

File: gdsfactory/tech.py (hunk 2 above)
Comment (simbilod) [reply gdsfactory/tech.py]: tech.py has been updated, good catches

**Comment by joamatab** [comment]
Looks great Simon,

Where can we read about mesh order?

Im assuming mesh order is like priority (the higher the number it overrides lower mesh orders)

**Comment by simbilod** [comment]
> Looks great Simon,
> 
> Where can we read about mesh order?
> 
> Im assuming mesh order is like priority (the higher the number it overrides lower mesh orders)

You would be right! I'll put together some nice-looking docs when I have a moment
